mscore/albummanager.cpp

```diff
@@ -295,19 +490,25 @@ void AlbumManager::startPlayback()
 void AlbumManager::updateScoreOrder(QModelIndex sourceParent, int sourceStart, int sourceEnd,
                                     QModelIndex destinationParent, int destinationRow)
 {
-    for (int i = 0; i < _items.size(); i++) {
-        for (int j = 0; j < _items.size(); j++) {
-            if (_items.at(j)->albumItem->score->title() != scoreList->item(j, 0)->text()) {
-                int h = scoreList->row(scoreList->findItems(_items.at(j)->albumItem->score->title(),
+    Q_UNUSED(sourceParent);
+    Q_UNUSED(sourceStart);
+    Q_UNUSED(sourceEnd);
+    Q_UNUSED(destinationParent);
+    Q_UNUSED(destinationRow);
+
+    for (int i = 0; i < int(m_items.size()); i++) {
+        for (int j = 0; j < int(m_items.size()); j++) {
+            if (m_items.at(j)->albumItem.score->title() != scoreList->item(j, 0)->text()) {
+                int h = scoreList->row(scoreList->findItems(m_items.at(j)->albumItem.score->title(),
                                                             Qt::MatchExactly).first());
-                std::swap(_items.at(j), _items.at(h));
+                std::swap(m_items.at(j), m_items.at(h));
                 break;
-            } else if (j == _items.size() - 1) {
-                goto end;
+            } else if (j == int(m_items.size()) - 1) {
+                goto exit_loops;
```

Comment:
I am always very suspicious of `goto`s. No good can come of it in my opinion.

I would personally have a variable just before the start of the inner loop called `lastIndex`. I would then set this variable to the value of `j` at the start of the inner loop. Then, after the inner loop, but still within the outer loop, I would compare this using your `(j == int(m_items.size()) - 1)` condition, breaking from the loop if it were true.

That said, if you really feel like `goto` is the best option, I trust you to use it sparingly and _very_ carefully.

Reply:
I only ever use `goto` for exiting nested loops. I find it a bit more elegant than adding a new variable and a new condition.